mmdet/models/dense_heads/yolo_head.py

```diff
@@ -257,20 +257,23 @@ def _get_bboxes_single(self,
             # (h, w, num_anchors*num_attrib) -> (h*w*num_anchors, num_attrib)
             pred_map = pred_map.permute(1, 2, 0).reshape(-1, self.num_attrib)
 
-            pred_map[..., :2] = torch.sigmoid(pred_map[..., :2])
-            bbox_pred = self.bbox_coder.decode(multi_lvl_anchors[i],
-                                               pred_map[..., :4], stride)
+            xy = torch.sigmoid(pred_map[..., :2])
+            xywh = torch.cat((xy, pred_map[..., 2:4]), axis=-1)
+            bbox_pred = self.bbox_coder.decode(multi_lvl_anchors[i], xywh,
+                                               stride)
+
             # conf and cls
             conf_pred = torch.sigmoid(pred_map[..., 4]).view(-1)
             cls_pred = torch.sigmoid(pred_map[..., 5:]).view(
                 -1, self.num_classes)  # Cls pred one-hot.
 
             # Filtering out all predictions with conf < conf_thr
             conf_thr = cfg.get('conf_thr', -1)
-            conf_inds = conf_pred.ge(conf_thr).nonzero().flatten()
-            bbox_pred = bbox_pred[conf_inds, :]
-            cls_pred = cls_pred[conf_inds, :]
-            conf_pred = conf_pred[conf_inds]
+            if conf_thr >= 0:
+                conf_inds = conf_pred > conf_thr
```

Comment:
In my experience with torch=1.6.0, it works as following:
`# conf_inds = conf_pred.ge(conf_thr).nonzero().flatten() # not work`
`mask = conf_pred >= conf_thr`
`conf_inds = mask.nonzero(as_tuple=False).squeeze(1)`

Reply:
I'll give it a shot. Actually, I think indexing with a boolean tensor and indexing with indicies from `nonzero` get converted to the same ONNX ops, so the main advantage of your code is that the nonzero-indices get re-used for multiple indexing operations.

Reply:
Totally agree. Bool indexing works with 1D, when it comes to 2D tensor, it may fail (my experience). See following warning: `UserWarning: Exporting aten::index operator with indices of type Byte. Only 1-D indices are supported. In any other case, this will produce an incorrect ONNX graph.
"Exporting aten::index operator with indices of type Byte`